beacon-chain/sync/service.go

```diff
@@ -80,13 +81,18 @@ func (ss *Service) Stop() error {
 // Status always returns nil.
 // TODO(1206): Add service health checks.
 func (ss *Service) Status() error {
+	if ss.failStatus != nil {
+		log.Errorf("Sync service is unhealthy : %v ", ss.failStatus)
+		return ss.failStatus
+	}
 	return nil
 }
 
 func (ss *Service) run() {
 	ss.Querier.Start()
 	synced, err := ss.Querier.IsSynced()
 	if err != nil {
+		ss.failStatus = err
```

Comment:
The server crashes on the next line so there is no point in setting this.

Reply:
removed this line